File: PHP/FindTheMaximumNumberOfElementsInSubset.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

class FindTheMaximumNumberOfElementsInSubset {    

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function maximumLength($nums) {
        $maxNum = max($nums);
        $count = [];

        foreach ($nums as $num) {
            $count[$num] = ($count[$num] ?? 0) + 1;
        }
        $ans = isset($count[1])
            ? $count[1] - (($count[1] % 2 == 0) ? 1 : 0)
            : 1;

        foreach ($nums as $num) {
            if ($num == 1) {
                continue;
            }
            $length = 0;
            $x = $num;

            while (
                $x <= $maxNum &&
                isset($count[$x]) &&
                $count[$x] >= 2
            ) {
                $length += 2;
                $x *= $x;
            }
            $ans = max($ans, $length + (isset($count[$x]) ? 1 : -1));
        }
        return $ans;
    }

}

File: GO/FindTheMaximumNumberOfElementsInSubset.go
/**
 * @version GO 1.26.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

func maximumLength(nums []int) int {
	maxNum := nums[0]

	for _, x := range nums {
		if x > maxNum {
			maxNum = x
		}
	}
	count := make(map[int]int)
	for _, num := range nums {
		count[num]++
	}
	ans := 1
	if c, ok := count[1]; ok {
		if c%2 == 0 {
			ans = c - 1
		} else {
			ans = c
		}
	}
	for _, num := range nums {
		if num == 1 {
			continue
		}
		length := 0
		x := int64(num)
		for x <= int64(maxNum) {
			c, ok := count[int(x)]
			if !ok || c < 2 {
				break
			}
			length += 2
			x *= x
		}
		if _, ok := count[int(x)]; ok {
			if length+1 > ans {
				ans = length + 1
			}
		} else {
			if length-1 > ans {
				ans = length - 1
			}
		}
	}
	return ans
}
